src/Admin/FieldDescription.php

```diff
@@ -27,7 +27,7 @@ public function setAssociationMapping(array $associationMapping): void
     {
         $this->associationMapping = $associationMapping;
 
-        $this->type = $this->type ?: $associationMapping['type'];
+        $this->type = $this->type ?: (string) $associationMapping['type'];
```

Comment:
so we keep this cast to string for now?

Reply:
For now, I would say yes. It works like this.